src/renderer/modules/common/flux.ts

```diff
@@ -183,34 +183,71 @@ const SnapshotStoreClass = await waitForModule<typeof SnapshotStore>(
   filters.bySource("SnapshotStores"),
 );
 
-export interface FluxHooks {
-  useStateFromStores: <T>(
-    stores: Store[],
-    callback: () => T,
-    deps?: React.DependencyList,
-    compare?: (a: T, b: T) => boolean,
-  ) => T;
-  statesWillNeverBeEqual: <T>(a: T, b: T) => boolean;
-  useStateFromStoresArray: <T>(
-    stores: Store[],
-    callback: () => T,
-    deps?: React.DependencyList,
-  ) => T;
-  useStateFromStoresObject: <T>(
-    stores: Store[],
-    callback: () => T,
-    deps?: React.DependencyList,
-  ) => T;
+function shallowEqual<T extends Record<string, unknown>>(a: T, b: T): boolean {
+  if (a === b) return true;
+
+  const keysA = Object.keys(a);
+  const keysB = Object.keys(b);
+
+  if (keysA.length !== keysB.length) return false;
+
+  for (let i = 0; i < keysA.length; i++) {
+    let key = keysA[i];
+    if (a[key] !== b[key]) return false;
+  }
+
+  return true;
```

Comment:
You can use lodash `xor` with the keys here - https://stackoverflow.com/a/54788640/6911703